### PR TITLE
HUM-449 EC auditor

### PR DESCRIPTION
**Pull request by corystone**
Iterate over all of the index db entries and check their shard md5s.
If they aren't right, quarantine the shard data and delete from db.
Write the object metadata to a quarantine marker
Dont quarantine nursery objects if we cant find them.
Start adding policy stuff to recon/repair

### CODE REVIEW

File: objectserver/auditor.go [hunk 1]
```diff
@@ -62,6 +67,77 @@ type Auditor struct {
...
+			return bytes, fmt.Errorf("Metadata missing ETag: %s", string(metabytes))
+		}
+
+		file, err := os.Open(path)
+		if err != nil {
+			return bytes, fmt.Errorf("Error opening file: %s", err)
+		}
+		defer file.Close()
+		h := md5.New()
+		bytes, err = common.Copy(file, h)
```
Comment (dpgoetz) [inline objectserver/auditor.go]: should be rate limit this? seems like we could read chunks and sleep a bit between. for a 5GB file would just blow through it here and then sleep a long time below?

File: objectserver/auditor.go [hunk 2]
```diff
@@ -62,6 +67,77 @@ type Auditor struct {
...
+
+	bytes := int64(0)
+	if !skipMd5 {
+		file, err := os.Open(path)
+		if err != nil {
+			return bytes, fmt.Errorf("Error opening file: %s", err)
+		}
+		defer file.Close()
+		h := md5.New()
+		bytes, err = common.Copy(file, h)
```
Comment (dpgoetz) [inline objectserver/auditor.go]: same as above

File: objectserver/ecobj.go [hunk 3]
```diff
@@ -317,72 +289,142 @@ func (o *ecObject) Replicate(prirep objectserver.PriorityRepJob) error {
...
+			failed[i] = node
+			readFails++
 			continue
 		}
 		bodies[i] = resp.Body
+		readSuccesses++
 	}
-	if toDeviceShard == -1 {
-		return fmt.Errorf("ToDevice %s:%d  %s is not in list of nodes for this object", prirep.ToDevice.Ip, prirep.ToDevice.Port, prirep.ToDevice.Device)
+	o.logger.Info("Reconstruct successful reads", zap.Int("successes", readSuccesses))
```
Comment (dpgoetz) [inline objectserver/ecobj.go]: prob Debug log

**Comment by dpgoetz** [comment]
still need to go through more. but couple initial things

File: objectserver/ecobj.go (hunk 3 above)
Comment (corystone) [reply objectserver/ecobj.go]: Yeah, this is just deleted.

**Pull request merged**